File: iconics-java/src/main/java/com/open/iconics/view/IconicsImageButton.java
package com.open.iconics.view;

import android.content.Context;
import android.util.AttributeSet;
import android.widget.ImageButton;

public class IconicsImageButton extends IconicsImageView {

    public IconicsImageButton(Context context) {
        this(context, null);
    }

    public IconicsImageButton(Context context, AttributeSet attrs) {
        this(context, attrs, android.R.attr.buttonStyle);
    }

    public IconicsImageButton(Context context, AttributeSet attrs, int defStyle) {
        super(context, attrs, defStyle);
        setFocusable(true);
    }

    @Override
    protected boolean onSetAlpha(int alpha) {
        return false;
    }

    @Override
    public CharSequence getAccessibilityClassName() {
        return ImageButton.class.getName();
    }
}

File: iconics-kotlin/src/main/java/com/open/iconics/view/IconicsImageButton.kt
package com.open.iconics.view

import android.content.Context
import android.util.AttributeSet
import android.widget.ImageButton

open class IconicsImageButton @JvmOverloads constructor(
    context: Context,
    attrs: AttributeSet? = null,
    defStyle: Int = android.R.attr.buttonStyle
) : IconicsImageView(context, attrs, defStyle) {

    init {
        isFocusable = true
    }

    override fun onSetAlpha(alpha: Int): Boolean = false

    override fun getAccessibilityClassName(): CharSequence = ImageButton::class.java.name
}
